File: FlexServ Backend/SpringBoot/BusinessService/src/main/java/com/flexserv/dto/request/AddressRequest.java
package com.flexserv.dto.request;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class AddressRequest {

    private String houseNo;
    private String street;
    
    @NotBlank(message = "Area is required")
    private String area;

    @NotBlank(message = "City is required")
    private String city;

    @NotBlank(message = "State is required")
    private String state;

    @NotBlank(message = "Pincode is required")
    @Pattern(regexp = "^[1-9][0-9]{5}$", message = "Enter a valid 6-digit Indian pincode")
    private String pincode;
}

File: FlexServ Backend/DotNet/AdminService/Dtos/AddressRequest.cs
namespace AdminService.Dtos
{
    public class AddressRequest
    {
        public string? HouseNo { get; set; }
        public string? Street { get; set; }
        public string? Area { get; set; }
        public string? City { get; set; }
        public string? State { get; set; }
        public string? Pincode { get; set; }
    }
}
